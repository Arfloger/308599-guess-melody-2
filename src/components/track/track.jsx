import React from 'react';
import PropTypes from 'prop-types';

export const Track = (props) => {
  const {value} = props;
  return (
    <div className="track">
      <button className="track__button track__button--play" type="button"></button>
      <div className="track__status">
        <audio></audio>
      </div>
      <div className="game__answer">
        <input
          className="game__input visually-hidden"
          type="checkbox"
          name="answer"
          value={value}
          id={value}
        />
        <label className="game__check" htmlFor={value}>Отметить</label>
      </div>
    </div>
  );
};

Track.propTypes = {
  value: PropTypes.string.isRequired,
};
