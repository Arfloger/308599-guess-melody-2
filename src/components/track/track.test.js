import React from "react";
import renderer from "react-test-renderer";

import {Track} from "./track.jsx";

it(`Track correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<Track
      value = {``}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
