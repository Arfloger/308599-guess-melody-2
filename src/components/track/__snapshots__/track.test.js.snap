// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track correctly renders after relaunch 1`] = `
<div
  className="track"
>
  <button
    className="track__button track__button--play"
    type="button"
  />
  <div
    className="track__status"
  >
    <audio />
  </div>
  <div
    className="game__answer"
  >
    <input
      className="game__input visually-hidden"
      id=""
      name="answer"
      type="checkbox"
      value=""
    />
    <label
      className="game__check"
      htmlFor=""
    >
      Отметить
    </label>
  </div>
</div>
`;
